root/static/scripts/edit/components/FormRowTextList.js

```diff
@@ -0,0 +1,134 @@
+/*
+ * @flow strict
+ * Copyright (C) 2024 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React, {useState} from 'react';
+
+import AddButton from './AddButton.js';
+import FieldErrors from './FieldErrors.js';
+import FormLabel from './FormLabel.js';
+import FormRow from './FormRow.js';
+import RemoveButton from './RemoveButton.js';
+
+component TextListRow(
+  name: string,
+  onChange: (event: SyntheticEvent<HTMLInputElement>) => void,
+  onRemove: (event: SyntheticEvent<HTMLInputElement>) => void,
+  removeButtonLabel: string,
+  value: string,
+) {
+  return (
+    <div className="text-list-row">
+      <input
+        className="value with-button"
+        name={name}
+        onChange={onChange}
+        type="text"
+        value={value}
+      />
+      <RemoveButton onClick={onRemove} title={removeButtonLabel} />
+    </div>
+  );
+}
+
+const initialRows = (repeatable: RepeatableFieldT<FieldT<string>>) => {
+  if (repeatable.field.length === 0) {
+    return [{name: repeatable.html_name + '.0', value: ''}];
+  }
+
+  return repeatable.field.map((field, index) => ({
+    name: repeatable.html_name + '.' + index,
+    value: field.value ?? '',
+  }));
+};
+
+component FormRowTextList(
+  addButtonLabel: string,
+  addButtonId: string,
+  label: string,
+  removeButtonLabel: string,
+  repeatable: RepeatableFieldT<FieldT<string>>,
+  required: boolean = false,
+) {
+  const newRow = (name: string, value: string, index: number) => {
+    return {name: name + '.' + index, value};
+  };
+
+  const [rows, setRows] = useState(initialRows(repeatable));
+
+  const addRow = () => {
+    const index = rows.length;
+
+    setRows([...rows, newRow(repeatable.html_name, '', index)]);
+  };
+
+  const changeRow = (index: number, value: string) => {
+    const newRows = [...rows];
+    newRows[index] = newRow(repeatable.html_name, value, index);
+    setRows(newRows);
+  };
+
+  const removeRow = (index: number) => {
+    if (rows.length === 1) {
+      setRows([newRow(repeatable.html_name, '', 0)]);
+      return;
+    }
+
+    const newRows = [...rows];
+    newRows.splice(index, 1);
+    setRows(newRows);
+  };
+
+  return (
+    <>
+      <FormLabel forField={repeatable} label={label} required={required} />
+
+      <div className="form-row-text-list">
+        {rows.map((field, index) => (
+          <TextListRow
+            key={index}
+            name={field.name}
+            onChange={(event) => changeRow(index, event.currentTarget.value)}
+            onRemove={() => removeRow(index)}
```

Comment:
One issue here is that ideally `TextListRow` should be memoized, and passing callbacks like this will break memoization. So:
1. `TextListRow` should be using `React.memo`
2. Any callback being passed to `TextListRow` should be using `useCallback`

However, having the `index` as a dependency is not a good idea, so this would actually be easier using `useReducer` + `dispatch`, or using the key to find the row instead of the index.